sfge/src/main/java/com/salesforce/graph/ops/MethodTypeMatchUtil.java

```diff
@@ -0,0 +1,249 @@
+package com.salesforce.graph.ops;
+
+import com.salesforce.apex.jorje.ASTConstants;
+import com.salesforce.exception.UnexpectedException;
+import com.salesforce.graph.Schema;
+import com.salesforce.graph.build.CaseSafePropertyUtil;
+import com.salesforce.graph.symbols.AbstractClassInstanceScope;
+import com.salesforce.graph.symbols.SymbolProvider;
+import com.salesforce.graph.symbols.apex.ApexClassInstanceValue;
+import com.salesforce.graph.symbols.apex.ApexValue;
+import com.salesforce.graph.vertex.*;
+import org.apache.tinkerpop.gremlin.process.traversal.P;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.salesforce.graph.ops.TypeableUtil.NOT_A_MATCH;
+
+public final class MethodTypeMatchUtil {
```

Comment:
No code changes. Only extracted type-matching related methods to a new class from `MethodUtil.java`.

Reply:
Will add javadoc.